network/message/message.go

```diff
@@ -1,33 +1,91 @@
 package message
 
-import "github.com/gladiusio/legion/utils"
-
-// New returns a new Message with the specified data
-func New(sender utils.LegionAddress, messageType string, body, checksum []byte, compression bool) *Message {
-	m := &Message{
-		Sender:      sender,
-		Type:        messageType,
-		Body:        body,
-		Compression: compression,
-	}
+import (
+	"errors"
+	"sync"
+
+	"github.com/gladiusio/legion/network/message/flatb"
+	"github.com/gladiusio/legion/utils"
+	flatbuffers "github.com/google/flatbuffers/go"
+)
 
-	m.calculateChecksum()
+// builderPool keeps a dynamically sized pool of builder objects
+var builderPool = sync.Pool{
```

Comment:
Why package scope?

Reply:
So that the allocation is done once and the pool is available to all new messages. When a user calls `message.New(...)` no matter what they'll have access to the same resource pool.